### PR TITLE
This threads a context.Context through the webhook interfaces we expose.

### PR DESCRIPTION
**Pull request by mattmoor**
Related to: https://github.com/knative/pkg/issues/306

/hold

I want to stage the PR to serving (and @n3wscott to Eventing) before I remove the hold

### CODE REVIEW

File: apis/interfaces.go [hunk 1]
```diff
@@ -17,28 +17,30 @@ limitations under the License.
...
+
 	authenticationv1 "k8s.io/api/authentication/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
 // Defaultable defines an interface for setting the defaults for the
 // uninitialized fields of this instance.
 type Defaultable interface {
-	SetDefaults()
+	SetDefaults(context.Context)
```
Comment (grantr) [inline apis/interfaces.go]: What is this context argument expected to be used for? Do we have an example?

Comment (vagababov) [reply apis/interfaces.go]: For example we have a request to make default revision timeout configurable by the operator.
ConfigMaps, attached to the context can be used to get the operator desired default.

**Pull request merged**